### DISCUSSION TITLE
How to use this on a binary with arguments?

### DISCUSSION THREAD

**Discussion by beedell-roke** [KDAB/hotspot#644] [Q&A] [answered]
How do I use https://github.com/KDAB/hotspot/blob/35871adeec5adc882f2134804058b1071886bcc5/README.md?plain=1#L151 with an application which has arguments? Can I just quote-enclose it as my shell defines? I ask because a basic test doesn't appear to work:

> ```log
> RokeJulianLockhart@sayw4i:~$ perf record --call-graph dwarf 'echo yes'
> Failed to collect 'cycles:Pu' for the 'echo yes' workload: No such file or directory
> ```

**Comment by milianw** [comment, marked as answer]
Remove the quotes, just do:

```
perf record --call-graph dwarf echo yes this works
```

If you want to make it more explicit, add two `--` inbetween:

```
perf record --call-graph dwarf -- echo yes this works
```
